theano/tensor/tests/test_subtensor.py

```diff
@@ -491,6 +491,8 @@ def test_ok_list(self):
                           # Test with TensorConstant index.
                           (rand(4, 2, 3),
                            theano.tensor.constant([3, 3, 1, 1, 2, 2, 0, 0])),
+                          # Test with non-contiguous input
+                          (rand(6, 2)[::2], [1, 0])
```

Comment:
The goal of the new test is to test when the graph generate a non-contiguous index, do we work correctly? Do we add the gpu_contiguous to assert we are contiguous in the graph?

This was the bug we had. I don't think this new test detect that. The first fix added an gpu_contiguous on the input, not just the index, but we looked together at the code and it wasn't requested. So you removed that from the PR. https://github.com/Theano/Theano/pull/5469/files only add gpu_contiguous for the index, not for the input.